repair/repair.cc

```diff
@@ -910,6 +916,7 @@ struct repair_options {
     static constexpr const char* END_TOKEN = "endToken";
     static constexpr const char* RANGES_PARALLELISM_KEY = "ranges_parallelism";
     static constexpr const char* SMALL_TABLE_OPTIMIZATION_KEY = "small_table_optimization";
+    static constexpr const char* REDUCE_READER_OPTIMIZATION = "mixed_shard_optimization";
```

Comment:
The name isn't updated

Reply:
IMO, the name like mixed_shard_workaround is not much better than mixed_shard_optimization. 
It is aligned with small_table_optimization. More importantly, I am looking into a solution to make it a real optimization.

Reply:
I've meant they aren't consistent: `REDUCE_READER_OPTIMIZATION` and `mixed_shard_optimization`